Rubberduck.Inspections/QuickFixes/AccessSheetUsingCodeNameQuickFix.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Linq;
+using Antlr4.Runtime;
+using Rubberduck.Inspections.Abstract;
+using Rubberduck.Inspections.Concrete;
+using Rubberduck.Inspections.Results;
+using Rubberduck.Parsing;
+using Rubberduck.Parsing.Grammar;
+using Rubberduck.Parsing.Inspections.Abstract;
+using Rubberduck.Parsing.Inspections.Resources;
+using Rubberduck.Parsing.VBA;
+
+namespace Rubberduck.Inspections.QuickFixes
+{
+    public class AccessSheetUsingCodeNameQuickFix : QuickFixBase
+    {
+        private readonly RubberduckParserState _state;
+
+        public AccessSheetUsingCodeNameQuickFix(RubberduckParserState state)
+            : base(typeof(SheetAccessedUsingStringInspection))
+        {
+            _state = state;
+        }
+
+        public override void Fix(IInspectionResult result)
+        {
+            var referenceResult = (IdentifierReferenceInspectionResult)result;
+            var sheet = ((VBAParser.IndexExprContext)referenceResult.Context.Parent.Parent).argumentList().argument(0);
+            // Get rid of leading and trailing quotes
+            var sheetName = string.Concat(sheet.GetText().Skip(1).Take(sheet.GetText().Length - 2));
```

Comment:
This is the sheet's `Name`, not necessarily its `CodeName`. I believe we need some Excel interop to figure out the correct sheet name to use here.

Reply:
Or, we need to access the properties of the `VBComponent` for that worksheet, and get the correct `(Name)`. In any case taking the string and turning it into an identifier isn't going to end well.

    Set sheet = ThisWorkbook.Worksheets("Summary Sheet - January")

Reply:
Note, I've no idea how we're going to go about mocking a VBComponent that's interpreted as an Excel worksheet - doing this right is likely going to make the inspection and quick-fix very hard to test.

Reply:
Wow, how did I miss that? 😕